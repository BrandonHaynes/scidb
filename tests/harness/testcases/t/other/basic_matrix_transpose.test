--setup
CREATE ARRAY E <x: int32> [ i=0:2,3,0, j=0:2,3,0]
CREATE ARRAY E1 <x: int32 > [ i=0:1,2,0, j=0:2,3,0]
CREATE ARRAY E2 <var:string, var1:int32> [i=0:4,2,0]
CREATE ARRAY E3 <var:string, var1:int32> [ i=0:2,3,0, j=0:2,3,0]
CREATE ARRAY E4 <var:string, var1:int32> [ i=0:2,3,0, j=0:2,1,0]

--test
store(build(E, i), E)
store(build(E1, i+j+1), E1)
store(join(build(<var:string> [i=0:4,2,0], string(i)+'z'),  build(<var:int32> [i=0:4,2,0], i+1)),E2)
store(join(build(<var:string> [ i=0:2,3,0, j=0:2,3,0], string(i)+string(j)),  build(<var:int32> [ i=0:2,3,0, j=0:2,3,0], i+j+1)),E3)
store(join(build(<var:string> [ i=0:2,3,0, j=0:2,1,0], string(i)+string(j)),  build(<var:int32> [ i=0:2,3,0, j=0:2,1,0], i+j+1)),E4)

transpose(E)
transpose(E1)
transpose(E2)
transpose(E3)
transpose(E4)


--cleanup
remove(E)
remove(E1)
remove(E2)
remove(E3)
remove(E4)
