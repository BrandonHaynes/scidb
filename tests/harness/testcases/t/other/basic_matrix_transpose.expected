Query was executed successfully

Query was executed successfully

Query was executed successfully

Query was executed successfully

Query was executed successfully

{i,j} x
{0,0} 0
{0,1} 0
{0,2} 0
{1,0} 1
{1,1} 1
{1,2} 1
{2,0} 2
{2,1} 2
{2,2} 2

{i,j} x
{0,0} 1
{0,1} 2
{0,2} 3
{1,0} 2
{1,1} 3
{1,2} 4

{i} var,var1
{0} '0z',1
{1} '1z',2
{2} '2z',3
{3} '3z',4
{4} '4z',5

{i,j} var,var1
{0,0} '00',1
{0,1} '01',2
{0,2} '02',3
{1,0} '10',2
{1,1} '11',3
{1,2} '12',4
{2,0} '20',3
{2,1} '21',4
{2,2} '22',5

{i,j} var,var1
{0,0} '00',1
{1,0} '10',2
{2,0} '20',3
{0,1} '01',2
{1,1} '11',3
{2,1} '21',4
{0,2} '02',3
{1,2} '12',4
{2,2} '22',5

{j,i} x
{0,0} 0
{0,1} 1
{0,2} 2
{1,0} 0
{1,1} 1
{1,2} 2
{2,0} 0
{2,1} 1
{2,2} 2

{j,i} x
{0,0} 1
{0,1} 2
{1,0} 2
{1,1} 3
{2,0} 3
{2,1} 4

{i} var,var1
{0} '0z',1
{1} '1z',2
{2} '2z',3
{3} '3z',4
{4} '4z',5

{j,i} var,var1
{0,0} '00',1
{0,1} '10',2
{0,2} '20',3
{1,0} '01',2
{1,1} '11',3
{1,2} '21',4
{2,0} '02',3
{2,1} '12',4
{2,2} '22',5

{j,i} var,var1
{0,0} '00',1
{0,1} '10',2
{0,2} '20',3
{1,0} '01',2
{1,1} '11',3
{1,2} '21',4
{2,0} '02',3
{2,1} '12',4
{2,2} '22',5

Query was executed successfully

Query was executed successfully

Query was executed successfully

Query was executed successfully

Query was executed successfully

